### PR TITLE
Added Getter Methods and Inserter Method to Data Interface

### PR DESCRIPTION
**Pull request by nguya03584**
## Context

Getter methods were added to allow us to extract data from the database. An inserter method was added to let us store the recommended car for a given accountholder to the database. 

## Your Changes

**Description**:

I have added 3 new methods to the CarDataInterface: 2 getters and 1 inserter. One of the getters is to get a car from the database using the given car name, and the other getter is to get all cars in the database. For the AccountHolderDataInterface, I have 2 getters. One getter will get the financial transactions of a client while the other get the most recent credit score of a client. Both getters will use a given client ID to look them up in the database. The changes made are related to issue #9.

**Type of change** (select all that apply):

- [x] New feature (additional functionality separate from existing code)

## Testing

The methods added were abstract methods, so extensive testing was not needed. The codes were able to compile, and no errors were shown by Intellij

## Checklist

- [x] Code is up to date with main branch (no merge conflicts)
- [x] Code has been properly tested
- [ ] Code has been properly documented
- [ ] One or two people have reviewed this pull request

### CODE REVIEW

**Review by Infinityworld13 — APPROVED** [review]

File: service/src/main/java/com/aviva/DataAccess/AccountHolderDataInterface.java [hunk 1]
```diff
@@ -10,4 +10,11 @@
     public AccountHolder getClientByID(String ID);
 
     public Collection<AccountHolder> getAllClients();
+
+    // Get the most recent credit score of the bankaccount holder with the given ID.
+    public String getClientMostRecentCreditScore(String ID);
+
+    // Get all of financial transaction (deposits and withdrawls)
+    // of the bankaccount holder with the given ID.
+    public JSONObject getClientFinancialTransaction(String ID);
```
Comment (leenlab2) [inline service/src/main/java/com/aviva/DataAccess/AccountHolderDataInterface.java:19]: You forgot to import JSONObject

File: service/src/main/java/com/aviva/DataAccess/CarDataInterface.java [hunk 2]
```diff
@@ -7,4 +7,10 @@
 
 public interface CarDataInterface {
     public ArrayList<Car> loadCarData(String file) throws IOException, ClassNotFoundException, InterruptedException;
+
+    public String getCarByName(String carName);
+
+    public JSONObject getAllCars();
```
Comment (leenlab2) [inline service/src/main/java/com/aviva/DataAccess/CarDataInterface.java:13]: You forgot to import JSONObject

File: service/src/main/java/com/aviva/DataAccess/CarDataInterface.java [hunk 3]
```diff
@@ -7,4 +7,10 @@
 
 public interface CarDataInterface {
     public ArrayList<Car> loadCarData(String file) throws IOException, ClassNotFoundException, InterruptedException;
+
+    public String getCarByName(String carName);
```
Comment (leenlab2) [inline service/src/main/java/com/aviva/DataAccess/CarDataInterface.java:11]: I think this method should return JSONObject

Comment (leenlab2) [inline service/src/main/java/com/aviva/DataAccess/CarDataInterface.java:11]: This should return a Car

**Comment by leenlab2** [comment]
I added some specific comments, but more generally, we should add proper javadoc comments to the methods

**Review by Alamgir-K — APPROVED** [review]
Everything looks good to me. Some things might need changing once I get started with the SQL code but we'll figure that out at a later point.

**Comment by leenlab2** [comment]
@Alamgir-K is going to update this interface in a separate PR

**Pull request closed without merging**